plugins/pulsar-records-storage/src/test/java/com/github/bsideup/liiklus/pulsar/config/PulsarRecordsStorageConfigurationTest.java

```diff
@@ -0,0 +1,93 @@
+package com.github.bsideup.liiklus.pulsar.config;
+
+import com.github.bsideup.liiklus.pulsar.config.PulsarRecordsStorageConfiguration.PulsarProperties;
+import com.github.bsideup.liiklus.pulsar.container.PulsarTlsContainer;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.apache.pulsar.client.impl.auth.AuthenticationBasic;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.utility.MountableFile;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class PulsarRecordsStorageConfigurationTest {
+
+    @Test
+    void shouldSupportTLS() throws Exception {
+        try (var pulsar = new PulsarTlsContainer()) {
+            pulsar.start();
+            var configuration = new PulsarRecordsStorageConfiguration();
+
+            var properties = new PulsarProperties();
+            properties.setServiceUrl(pulsar.getPulsarBrokerUrl());
+
+            var topicName = UUID.randomUUID().toString();
+
+            try (var client = configuration.createClient(properties)) {
+                assertThatThrownBy(() -> client.getPartitionsForTopic(topicName).get(10, TimeUnit.SECONDS))
```

Comment:
could be TestFactory + dynamicTest, so that it will be shown as a separate case

Reply:
too heavy, since each test would require a new container

Reply:
also, this is an intended check that what we're going to check only works when everything is configured correctly (the test itself) and to avoid false positives

Reply:
no, thats the point, you could prepare everything and then spawn only checks, so all heavy lifting will be done once

Reply:
ah, wait, mb won't work within another try catch